pkg/tasks/prometheus.go

```diff
@@ -36,235 +38,253 @@ func NewPrometheusTask(client *client.Client, factory *manifests.Factory, config
 	}
 }
 
-func (t *PrometheusTask) Run(ctx context.Context) error {
+func (t *PrometheusTask) Run(ctx context.Context) StateErrors {
+	b := &client.StateErrorBuilder{}
+
+	prometheus, err := t.create(ctx)
+	b.AddError(client.DegradedState, err)
+
+	if prometheus == nil {
```

Comment:
if `t.create` fails because the API returned an unexpected error then we would skip validation and the cluster operator status might be incomplete?
While it's an edge case, it would be better not to rely on the returned result. IIUC the validation only needs the name/namespace of the Prometheus object which are known in advance.

Reply:
+1, I will handle this